addons/mrp/static/src/js/mrp.js

```diff
@@ -94,14 +98,16 @@ var FieldPdfViewer = FieldBinaryFile.extend({
      */
     on_file_change: function (ev) {
         this._super.apply(this, arguments);
+        var files = ev.target.files;
+        if (!files || files.length === 0) {
+            return;
+        }
+        var fileURI = URL.createObjectURL(files[0]);
         if (this.PDFViewerApplication) {
-            var files = ev.target.files;
-            if (!files || files.length === 0) {
-              return;
-            }
-            var file = files[0];
             // TOCheck: is there requirement to fallback on FileReader if browser don't support URL
```

Comment:
is this referring to the fallback in your `else`? If it is, maybe remove this comment?

Reply:
I would guess this is referring to a browser that does not support:

`new File(['hello'], 'world')`

so this would still have a reason to be (for the if and the else)